crates/sui/tests/onsite_reconfig_observer_tests.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use prometheus::Registry;
+use sui_core::authority_aggregator::AuthAggMetrics;
+use sui_core::quorum_driver::reconfig_observer::OnsiteReconfigObserver;
+use sui_core::quorum_driver::reconfig_observer::ReconfigObserver;
+use sui_core::safe_client::SafeClientMetricsBase;
+use test_utils::authority::{spawn_test_authorities_with_fullnodes, test_authority_configs};
+use test_utils::network::wait_for_nodes_transition_to_epoch;
+
+#[tokio::test]
+async fn test_onsite_reconfig_observer_basic() {
+    telemetry_subscribers::init_for_testing();
+    let (authorities, fullnodes) =
+        spawn_test_authorities_with_fullnodes([].into_iter(), &test_authority_configs(), 1).await;
+    let fullnode = &fullnodes[0];
+    let qd = fullnode.with(|node| {
+        let qd = node
+            .transaction_orchestrator()
+            .unwrap()
+            .clone_quorum_driver();
+        assert_eq!(qd.current_epoch(), 0);
+        qd
+    });
+    let qd_clone = qd.clone_quorum_driver();
+    let registry = Registry::new();
+    let mut observer: OnsiteReconfigObserver = fullnode
+        .with_async(|node| async {
+            let rx = node.subscribe_to_epoch_change().await;
+            OnsiteReconfigObserver::new(
+                rx,
+                node.clone_authority_store(),
+                node.clone_committee_store(),
+                SafeClientMetricsBase::new(&registry),
+                AuthAggMetrics::new(&registry),
+            )
+        })
+        .await;
+
+    let _observer_handle = tokio::task::spawn(async move { observer.run(qd_clone).await });
```

Comment:
this may result in strange behavior if this test is moved to sim_test. You are pulling a clone of the fullnode's quorum driver out of the simulator node in which it was created, and then starting a task in the client node. Instead, I would start this task inside of the `with` handler on line 18 so that you don't need to exfiltrate the qd from the node that owns it.

Reply:
This is documented here, btw: https://github.com/MystenLabs/sui/tree/main/crates/sui-simulator#suinodehandle

Reply:
I updated the test case to use `with_async`.
However the fullnode builder, which grabs localhost is currently not compatible with sim-test, and it fails [here](https://github.com/MystenLabs/sui/blob/main/crates/test-utils/src/authority.rs#L72) due to IP conflict.  Will fix this in a separate PR and convert this to a sim-test

Reply:
fullnode builder is used extensively in simtest - let me see if i can figure out what's causing your problem

Reply:
yes your fix works, thank you!